save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkContestProjectController.kt

```diff
@@ -58,29 +50,80 @@ class LnkContestProjectController(
     fun getRatingsInContest(
         @PathVariable contestName: String,
         authentication: Authentication,
-    ): Flux<ContestResult> = Flux.fromIterable(
-        lnkContestProjectService.getByContestName(contestName)
-    )
+    ): Flux<ContestResult> = Flux.fromIterable(lnkContestProjectService.getAllByContestName(contestName))
         .map {
-            it.toContestResult()
+            it to lnkContestExecutionService.getBestScoreOfProjectInContestWithName(it.project, it.contest.name)
+        }
+        .filter { (_, score) ->
+            score != null
+        }
+        .map { (lnkContestProject, score) ->
+            lnkContestProject.toContestResult(score)
         }
 
     /**
      * @param organizationName
      * @param projectName
-     * @param numberOfContests
-     * @return best [numberOfContests] contests of a project with name [projectName]
+     * @param amount
+     * @param authentication
+     * @return best [amount] contests of a project with name [projectName]
      */
     @GetMapping("/{organizationName}/{projectName}/best")
     @PreAuthorize("permitAll()")
-    fun getBestProjectsContests(
+    fun getBestProjectContests(
         @PathVariable organizationName: String,
         @PathVariable projectName: String,
-        @RequestParam(defaultValue = "4") numberOfContests: Int,
+        @RequestParam(defaultValue = "4") amount: Int,
+        authentication: Authentication,
     ): Flux<ContestResult> = Flux.fromIterable(
-        lnkContestProjectService.getBestContestsByProject(projectName, organizationName, numberOfContests)
+        lnkContestProjectService.getByProjectNameAndOrganizationName(projectName, organizationName, amount)
+    )
+        .map {
```

Comment:
these operations are similar to operations in previous method -- can we extract to common private method?

Reply:
Will do it